editor/blocks/heading-block/index.js

```diff
@@ -0,0 +1,23 @@
+const { query, html, prop } = wp.blocks.query;
+
+wp.blocks.registerBlock( 'wp/heading', {
+	title: 'Heading',
+	icon: 'heading',
+
+	attributes: {
+		value: query( 'h1,h2,h3,h4,h5,h6', html() ),
+		headingType: query( 'h1,h2,h3,h4,h5,h6', prop( 'nodeName' ) ),
+	},
+
+	edit( attributes, onChange ) {
+		return wp.element.createElement( wp.blocks.EditableHeading, {
+			value: attributes.value,
+			headingType: attributes.headingType,
+			onChange: ( value ) => onChange( { value } )
+		} );
+	},
+
+	save( attributes ) {
+		return wp.element.createElement( 'p', null, attributes.value );
```

Comment:
Would want to do similar logic as above to create element as heading. Perhaps this could be pulled out to a function at the top of the file:

```js
function Heading( { nodeName = 'H1', children } ) {
	return wp.element.createElement( nodeName.toLowerCase(), null, children );
}

// ...

	save( attributes ) {
		const { headingType, value } = attributes;
		return <Heading nodeName={ headingType }>{ value }</Heading>;
	}
```